### PR TITLE
Code refactor for service delete method

### PR DESCRIPTION
**Pull request by sudhirverma**

### CODE REVIEW

File: src/openshift/service.ts [hunk 1]
```diff
@@ -51,24 +51,18 @@ export class Service extends OpenShiftItem {
...
-        let service: OpenShiftObject = treeItem;
-        if (!service) {
-            project = await vscode.window.showQuickPick(Service.odo.getProjects(), {placeHolder: "From which Project you want to delete Service"});
-            application = await vscode.window.showQuickPick(Service.odo.getApplications(project), {placeHolder: "From which Application you want to delete Service"});
-            if (application) {
-                service = await vscode.window.showQuickPick(Service.odo.getServices(application), {placeHolder: "Select Service to delete"});
-            }
-        }
+        let service: OpenShiftObject;
+        service = await Service.getOpenShiftCmdData(treeItem,
```
Comment (dgolovin) [inline src/openshift/service.ts]: this should be ```const application = await Service.getOpenShiftCmdData(treeItem,```

File: src/openshift/service.ts [hunk 2]
```diff
@@ -51,24 +51,18 @@ export class Service extends OpenShiftItem {
...
+            "From which Project you want to delete Service",
+            "From which Application you want to delete Service"
+        );
+        if (!treeItem) service = await vscode.window.showQuickPick(Service.getServiceNames(service), {placeHolder: "Select Service to delete"});
```
Comment (dgolovin) [inline src/openshift/service.ts]: ```if (!application) service = await vscode.window.showQuickPick(Service.getServiceNames(application), {placeHolder: "Select Service to delete"});```

Comment (sudhirverma) [reply src/openshift/service.ts]: `!application` always return false

File: src/openshift/service.ts (hunk 1 above)
Comment (dgolovin) [reply src/openshift/service.ts]: I am confused. First you call 
``` TypeScript
Service.getOpenShiftCmdData(treeItem, 
    "From which Project you want to delete Service",
    "From which Application you want to delete Service"
);
```
and it passes through treeItem, or returning application istance or null.
Why variable is called `service` then?

File: src/openshift/service.ts (hunk 2 above)
Comment (dgolovin) [reply src/openshift/service.ts]: After 
``` TypeScript
Service.getOpenShiftCmdData(treeItem, 
    "From which Project you want to delete Service",
    "From which Application you want to delete Service"
);
```
`application` variable would be ether the same `treeItem` or application instance if `treeItem` is `null | undefined` or `null` if quickpick was canceled.

Comment (dgolovin) [reply src/openshift/service.ts]: Your code would probably work, I have not tested it, but it just look confusing to me at first sight.

**Comment by sudhirverma** [comment]
testPR

**Pull request merged**